src/scripts/install_machine.py

```diff
@@ -0,0 +1,324 @@
+            content += "  expire: False\n"
+
+            user_data_file = tempfile.NamedTemporaryFile(
+                prefix="cockpit-machines-",
+                suffix=("-user-data"),
+                mode='w+'
+            )
+            user_data_file.write(content)
+            user_data_file.flush()
+            params.append(f"user-data={user_data_file.name}")
+
+    yield params
+
+
+def prepare_installation_source(params, args):
+    only_define = False
+    if args['type'] == 'create':
+        if not args['startVm']:
+            only_define = True
+            params.append("--print-xml")
+
+    if args['sourceType'] == "pxe":
+        params += ['--pxe', '--network', args['source']]
+    elif args['sourceType'] == "os":
+        params += ['--install', f"os={args['os']}"]
+    elif args['sourceType'] in ['disk_image', 'cloud']:
+        params.append("--import")
+    elif ((args['source'][0] == '/' and os.path.isfile(args['source'])) or
+            (args['sourceType'] == 'url' and args['source'].endswith(".iso"))):
+        if only_define:
+            return
+        params += ['--cdrom', args['source']]
+    else:
+        if only_define:
+            return
+        params += ['--location', args['source']]
+
+
+@contextmanager
+def prepare_virt_install_params(args):
+    logging.debug(args)
+    with prepare_unattended(args) as unattended_params, prepare_cloud_init(args) as cloud_init_params:
+        params = [
+            "virt-install",
+            "--connect", f"qemu:///{args['connectionName']}",
+            "--quiet",
+            "--name", args['vmName'],
+            "--os-variant", args['os'],
+            "--memory", str(args['memorySize']),
+        ]
+
+        if args['type'] == 'install' or ('storagePool' in args and args['storagePool'] != 'NewVolume'):
+            params += ["--check", "path_in_use=off"]
+
+        if args['sourceType'] != 'disk_image':
+            params += ["--wait", "-1"]
+
+        if args['type'] == 'install' or args['startVm']:
+            params.append("--noautoconsole")
+
+        # Disks
+        if args['type'] == 'install':
+            disks = args['disks'] or ['none']
+            for disk in disks:
+                params += ['--disk', disk]
+        else:
+            params.append("--disk")
+            if args['sourceType'] == "cloud":
+                backing_store = f",backing_store={args['source']},"
+            else:
+                backing_store = ""
+
+            if args['sourceType'] == 'disk_image':
+                params.append(f"{args['source']},device=disk")
+            elif args['storagePool'] == 'NoStorage':
+                params.append("none")
+            elif args['storagePool'] != 'NewVolume':
+                params.append(f"vol={args['storagePool']}/{args['storageVolume']}{backing_store}")
+            else:
+                params.append(f"size={args['storageSize']},format=qcow2{backing_store}")
```

Comment:
... do something like this

```suggestion
            else:
                if args['storagePool'] != 'NewVolume':
                    params.append(f"vol={args['storagePool']}/{args['storageVolume']}")
                else:
                    params.append(f"size={args['storageSize']},format=qcow2")

                if args['sourceType'] == "cloud":
                    params.append(",backing_store={args['source']}")
```

?

Reply:
This needs to be in a variable together, so that subprocess can process is properly. So I applied this fix with the change that I prestore it in a variable